src/foam/core/AbstractFObject.java

```diff
@@ -171,6 +166,15 @@ public boolean hasDefaultValue(String prop) {
         prop.updateDigest(this, md);
       }
 
+      // no chaining so return digest
+      if ( hash == null || hash.length == 0 ) {
+        return md.digest();
+      }
+
+      // calculate digest, update with previous hash and current hash
+      byte[] digest = md.digest();
+      md.update(hash);
```

Comment:
This looks like it is using the same hash object for the hashing together of the old hash and new message digest. To compute H(H(old_tail), H(new_object)) you probably need a new message digest instance.

Reply:
According to the JavaDocs, calling `digest` resets the MessageDigest instance allowing it to be reused again.

Reply:
Fair enough. That's different to most of the hash libraries I see but if it works that way then you're good.